modules/lymphgen/1.0/lymphgen.smk

```diff
@@ -0,0 +1,385 @@
+        cnv_arm = str(rules._lymphgen_input_cnv.output.cnv_arm)
+    output:
+        result = CFG["dirs"]["lymphgen_run"] + "{outprefix}.results.{cnvs_wc}.{sv_wc}.tsv"
+    log:
+        stderr = CFG["logs"]["lymphgen_run"] + "{outprefix}.lymphgen.{cnvs_wc}.{sv_wc}.stderr.log",
+        stdout = CFG["logs"]["lymphgen_run"] + "{outprefix}.lymphgen.{cnvs_wc}.{sv_wc}.stdout.log"
+    params:
+        lymphgen_path = CFG["options"]["lymphgen_run"]["lymphgen_path"]
+    conda:
+        "envs/r-optparse-1.6.yaml"
+    wildcard_constraints:
+        cnvs_wc = "with_cnvs"
+    shell:
+        op.as_one_line("""
+        Rscript {params.lymphgen_path} -m {input.mutation_flat} -s {input.sample_annotation} -g {input.gene_list} -c {input.cnv_flat}
+        -a {input.cnv_arm} -o {output.result} > {log.stdout} 2> {log.stderr} """)
+
+# No CNVs
+rule _lymphgen_run_no_cnv:
+    input:
+        sample_annotation = _get_sample_annotation,
+        mutation_flat = str(rules._lymphgen_input_no_cnv.output.mutation_flat),
+        gene_list = str(rules._lymphgen_input_no_cnv.output.gene_list)
+    output:
+        result = CFG["dirs"]["lymphgen_run"] + "{outprefix}.results.{cnvs_wc}.{sv_wc}.tsv"
+    log:
+        stderr = CFG["logs"]["lymphgen_run"] + "{outprefix}.lymphgen.{cnvs_wc}.{sv_wc}.stderr.log",
+        stdout = CFG["logs"]["lymphgen_run"] + "{outprefix}.lymphgen.{cnvs_wc}.{sv_wc}.stdout.log"
+    params:
+        lymphgen_path = CFG["options"]["lymphgen_run"]["lymphgen_path"]
+    conda:
+        "envs/r-optparse-1.6.yaml"
+    wildcard_constraints:
+         cnvs_wc = "no_cnvs"
+    shell:
+        op.as_one_line("""
+        Rscript {params.lymphgen_path} -m {input.mutation_flat} -s {input.sample_annotation} -g {input.gene_list}
+        -o {output.result} > {log.stdout} 2> {log.stderr}""")
+
+
+# Symlinks the final output files into the module results directory (under '99-outputs/')
+rule _lymphgen_output_txt:
+    input:
+        txt = str(rules._lymphgen_run_cnv.output.result)
+    output:
+        txt = CFG["dirs"]["outputs"] + "{outprefix}.lymphgen_calls.{cnvs_wc}.{sv_wc}.tsv"
+    run:
+        op.relative_symlink(input.txt, output.txt)
+
+
+# Generates the target sentinels for each run, which generate the symlinks
+
+# Set the applicable wildcards, based on the provided input files
+# Are we running LymphGen with CNV data?
+if "sample_seg" in CFG["inputs"] and CFG["inputs"]["sample_seg"] != "" and CFG["inputs"]["sample_seg"] != "None":
+    cnvs_wc = ["with_cnvs", "no_cnvs"]
+else:
+    cnvs_wc = ["no_cnvs"]
+
+if "sv_info_tsv" in CFG["inputs"] and CFG["inputs"]["sv_info_tsv"] != "" and CFG["inputs"]["sample_seg"] != "None":
+    sv_wc = ["with_sv", "no_sv"]
+else:
+    sv_wc = ["no_sv"]
+
+rule _lymphgen_all:
+    input:
+        expand(
+            [
+                str(rules._lymphgen_output_txt.output.txt),
+            ],
+            outprefix = outprefix,
+            cnvs_wc = cnvs_wc,
+            sv_wc = sv_wc
```

Comment:
By default this generates all possible variants of the provided inputs. i.e. if you just provide SNVs, it will only generate one output (no_cnvs, no_svs), whereas if you provide SNVs and CNVs it will run LymphGen with both variants (with_cnvs, no_svs + no_cnvs, no_svs)